react/features/token-data/middleware.js

```diff
@@ -0,0 +1,69 @@
+import { SET_ROOM_URL } from '../app';
+import { getConfigParamsFromUrl } from '../base/config';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { setTokenData } from './actions';
+import { TokenData } from './TokenData';
+
+/**
+ * Middleware that parsing token data after setting new room URL.
+ *
+ * @param {Store} store - The Redux store.
+ * @returns {Function}
+ * @private
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case SET_ROOM_URL:
+        return _parseTokenData(store, next, action);
+    }
+
+    return next(action);
+});
+
+/**
+ * Parses token data from room URL and if it's found stores into Redux
+ * store.
+ *
+ * @param {Store} store - The Redux store in which the specified action is being
+ * dispatched.
+ * @param {Dispatch} next - The Redux dispatch function to dispatch the
+ * specified action to the specified store.
+ * @param {Action} action - The Redux action SET_ROOM_URL which is being
+ * dispatched in the specified store.
+ * @returns {Object} The new state that is the result of the reduction of the
+ * specified action.
+ * @private
+ */
+function _parseTokenData(store, next, action) {
+    const nextState = next(action);
+
+    const { config } = nextState['features/base/lib-jitsi-meet'];
+    const { roomUrl } = nextState['features/base/conference'];
+    const params = getConfigParamsFromUrl(roomUrl, true, 'search');
+    const token = params.jwt;
+    const isGuest = !config.enableUserRolesBasedOnToken;
+    const parsedToken = new TokenData(isGuest, token);
+    const {
+        issuer,
+        payload
+    } = parsedToken;
+
+    // Use JWT param as token if there is not other token set and if
+    // the iss field is not anonymous. If you want to pass data with
+    // JWT token but you don't want to pass the JWT token for
+    // verification the iss field should be set to "anonymous".
+    if (payload && issuer !== 'anonymous') {
+        parsedToken.token = token;
+    }
+
+    // If token was set in config (e.g. via URL params or deployment config
+    // injected in index.html) then let's use its value in token data.
+    if (config.token) {
```

Comment:
remove this, it shouldn't be needed. config will never have a token, that's wrong, and if there was a token in the URL we got it already.